File: docs/src/rust/user-guide/io/aws.rs
"""
# --8<-- [start:bucket]
use aws_sdk_s3::Region;

use aws_config::meta::region::RegionProviderChain;
use aws_sdk_s3::Client;
use std::borrow::Cow;

use polars::prelude::*;

#[tokio::main]
async fn main() {
    let bucket = "<YOUR_BUCKET>";
    let path = "<YOUR_PATH>";

    let config = aws_config::from_env().load().await;
    let client = Client::new(&config);

    let req = client.get_object().bucket(bucket).key(path);

    let res = req.clone().send().await.unwrap();
    let bytes = res.body.collect().await.unwrap();
    let bytes = bytes.into_bytes();

    let cursor = std::io::Cursor::new(bytes);

    let df = CsvReader::new(cursor).finish().unwrap();

    println!("{:?}", df);
}
# --8<-- [end:bucket]
"""

File: docs/src/python/user-guide/io/aws.py
"""
# --8<-- [start:read_parquet]
import polars as pl

bucket = "<YOUR_BUCKET>"
path = "<YOUR_PATH>"

df = pl.read_parquet(f"s3://{bucket}/{path}")
# --8<-- [end:bucket]

# --8<-- [start:scan_parquet]
import polars as pl

bucket = "<YOUR_BUCKET>"
path = "<YOUR_PATH>"

df = pl.scan_parquet(f"s3://{bucket}/{path}")
# --8<-- [end:scan_parquet]

# --8<-- [start:scan_parquet_query]
import polars as pl

bucket = "<YOUR_BUCKET>"
path = "<YOUR_PATH>"

df = pl.scan_parquet(f"s3://{bucket}/{path}").filter(pl.col("id") < 100).select("id","value")
# --8<-- [end:scan_parquet_query]


"""
